pandas/tests/series/test_combine_concat.py

```diff
@@ -181,7 +181,8 @@ def test_concat_empty_series_dtypes(self):
         # categorical
         assert pd.concat([Series(dtype='category'),
                           Series(dtype='category')]).dtype == 'category'
-        assert pd.concat([Series(dtype='category'),
+        # GH 18515
+        assert pd.concat([Series(np.array([]), dtype='category'),
```

Comment:
Why do you change this? (it's only the dtype of the category, it is still a categorical series, so for the concat that does not matter)

Reply:
Yes, the dtype of the category does matter for concat (and rightly so, since conceptually the fact that a categorical of ints is really a categorical is only an implementation detail when you're going to concat it to a non-categorical anyway).

(... or just try that test before and after my change)

Reply:
How does this PR change the behaviour of empty category series?

Reply:
I get that when #18515 is fixed, this test should be changed this way, but please leave that change for a PR that actually fixes #18515

Reply:
Is it clear to you that that test fails with _this_ PR? If yes, please clarify/reformulate what you're asking for.

Reply:
By the way: a categorical Series initialized with an empty Series (as opposed to no data at all) already has the right dtype.

Reply:
Well, my question is: 

> How does this PR change the behaviour of empty category series?

Reply:
If this PR changes (fixes) the dtype of categories for `pd.Series(dtype='category')`, please specify so and add an explicit test for this (and a whatsnew bug fix note).

Reply:
So `data is None` is handled with `data = {}`, and thus this PR affects this behaviour? So this does fix https://github.com/pandas-dev/pandas/issues/18515

Reply:
OK, asking is good. This PR makes (incidentally, but it is the correct behaviour) category series initialized with an empty ``dict`` behave like category series initialized with an empty ``list``, that is have ``object`` dtype. This fixes #18515 (but not  #17261 ). Since this test is about concatenating an empty float category ``Series`` to an empty float (non-category) ``Series``, I had to fix the former so that it still had dtype ``float64``.

So since you reopened #18515, now I will add "closes #18515" and push again.

OK?

Reply:
Yep. 
And in general, please add commits instead of amending if you make such additional changes. Makes reviewing easier.

Reply:
done, ping